File: src/app/conjugation/forms/causative-passive-affirmative.ts
import {Conjugation, Word, WordType} from "../conjugation";
import {PassiveAffirmative} from "./passive-affirmative";
import {CausativeAffirmative} from "./causative-affirmative";

export class CausativePassiveAffirmative implements Conjugation {
    getConjugation(word: Word): (Word | undefined) {

        const originalType = word.wordType;
        const causative = new CausativeAffirmative().getConjugation(word);
        if (causative === undefined) {
            return undefined;
        }

        return new PassiveAffirmative().getConjugation(causative.changeType(WordType.GodanVerb))?.changeType(originalType)
    }

    getTitle(): string {
        return "Causative-Passive, affirmative"
    }

}

File: backend/app/conjugation/forms/causative_passive_affirmative.py
from ..core import Conjugation, Word, WordType
from .causative_affirmative import CausativeAffirmative
from .passive_affirmative import PassiveAffirmative


class CausativePassiveAffirmative(Conjugation):

    title = 'Causative-Passive, affirmative'
    settings_title = 'Affirmative'

    def conjugate(self, word: Word) -> Word | None:
        original_type = word.word_type
        causative = CausativeAffirmative().conjugate(word)
        if causative is None:
            return None

        passive = PassiveAffirmative().conjugate(
            causative.change_type(WordType.GODAN_VERB)
        )
        if passive is None:
            return None

        return passive.change_type(original_type)
